chain/client/src/client_actor.rs

```diff
@@ -348,13 +348,13 @@ impl Handler<NetworkClientMessages> for ClientActor {
                 }
 
                 for shards_to_download in shards_to_download_vec {
```

Comment:
this vec seems error prone, suggestion to either keep only either statesync and state_sync_infos.
We can merge this hashmaps and assert that there is no common keys.
and then if there is missing `shard_id` in this hashmap - we should show the error / possibly ban the peer.